rust/kernel/thread.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::bindings;
+use crate::c_types;
+
+use alloc::boxed::Box;
+use core::ops::FnOnce;
+
+#[no_mangle]
+unsafe extern "C" fn rust_thread_func(data: *mut c_types::c_void) -> c_types::c_int {
+    Box::from_raw(data as *mut Box<dyn FnOnce()>)();
+    0
+}
+
+pub struct Thread {
+    task: *mut bindings::task_struct,
+}
+
+impl Thread {
+    pub fn new<F>(f: F, name: &str) -> Self
```

Comment:
I think this needs to be a CStr, rust strings are not nul terminated so just passing a pointer is unsafe.

Reply:
Done in updated version.